### ISSUE TITLE
Fix achievement IDs and instance difficulties that were changed in 7.3.5

### ISSUE THREAD

**Issue by SacredDuckwhale** [SacredDuckwhale/Rarity#11]
[CurseForge Link](https://www.curseforge.com/wow/addons/rarity?comment=1333)

> I tried for Mimiron's head today but Rarity didn't show an attempt at it.  I am pretty sure I tried it correctly...i pushed the button on the Mimiron fight and didnt talk to keepers.  It popped an attempt at getting Glittering Ball of Yarn from Auriaya when I looted her but nothing from Yog.

I think this only affects Ulduar, for now.

**Comment by SacredDuckwhale** [comment]
This should be fixed with the next release, but there's a few possible issues:
* Timewalking difficulty will never add attempts for Mimiron's Head, as there's no inherent way to make sure that the attempt will be eligible. I didn't just want to assume everybody would do 0 Keepers on anything but Normal difficulty
* I'm not sure why the pets use NPC detection and not the BOSS method, but I kept it for consistency and added a workaround for them instead of changing their method of detection. It's possible that this affects the defeat detection as well, but I'll have to look into it later ( when tackling #15)
* For Timewalking difficulty, I don't know if the drop rates are shared or if the items actually drop at all. My best bet is "no" and "yes", which may or may not turn out to be incorrect
* The workaround involves listening to the ENCOUNTER_END event to detect successful raid boss kills. I wasn't sure whether this or BOSS_KILL is the right one (both should work, for now), but since BOSS_KILL was labeled obsolete/removed on wowpogramming.com and I'm not sure if it fires for every encounter, I picked the other one
* I kept the old statistics data so it would be tracked. This should add new attempts on top of the old 10/25 man kills, but it won't update them, which means if they were to be summed up then the total amount would be off by the number of post 7.3.2-kills that can't be tracked like this. I don't think this is a problem in most cases, but it's something to keep in mind if those kills were to be added up later on
